apps/backend/db_patches/0148_CreateUpdateUserIdFunc.sql

```diff
@@ -0,0 +1,100 @@
+          --Delete TABLES row with OLD USER ID that is overlap with NEW USER ID  
+          UPDATE FAP_REVIEWERS SET USER_ID = P_NEW_USER_ID WHERE USER_ID = P_OLD_USER_ID 
+            AND FAP_ID NOT IN ( SELECT FAP_ID FROM  FAP_REVIEWERS WHERE USER_ID = P_NEW_USER_ID );
+          DELETE FROM FAP_REVIEWERS WHERE USER_ID = P_OLD_USER_ID;
+
+          UPDATE FAP_REVIEWS SET USER_ID = P_NEW_USER_ID WHERE USER_ID = P_OLD_USER_ID 
+            AND PROPOSAL_PK NOT IN ( SELECT PROPOSAL_PK FROM  FAP_REVIEWS WHERE USER_ID = P_NEW_USER_ID );
+          DELETE FROM FAP_REVIEWS WHERE USER_ID = P_OLD_USER_ID;
+
+          UPDATE INSTRUMENT_HAS_SCIENTISTS SET USER_ID = P_NEW_USER_ID WHERE USER_ID = P_OLD_USER_ID 
+            AND INSTRUMENT_ID NOT IN ( SELECT INSTRUMENT_ID FROM INSTRUMENT_HAS_SCIENTISTS WHERE USER_ID = P_NEW_USER_ID ) ;
+          DELETE FROM INSTRUMENT_HAS_SCIENTISTS WHERE USER_ID = P_OLD_USER_ID;
+
+          UPDATE PROPOSAL_USER SET USER_ID = P_NEW_USER_ID WHERE USER_ID = P_OLD_USER_ID 
+            AND PROPOSAL_PK NOT IN ( SELECT PROPOSAL_PK FROM PROPOSAL_USER WHERE USER_ID = P_NEW_USER_ID );
+          DELETE FROM PROPOSAL_USER WHERE USER_ID = P_OLD_USER_ID;
+
+          UPDATE ROLE_USER SET USER_ID = P_NEW_USER_ID WHERE USER_ID = P_OLD_USER_ID 
+            AND ROLE_ID NOT IN ( SELECT ROLE_ID FROM ROLE_USER WHERE USER_ID = P_NEW_USER_ID );
+          DELETE FROM ROLE_USER WHERE USER_ID = P_OLD_USER_ID;
+
+          UPDATE VISITS_HAS_USERS SET USER_ID = P_NEW_USER_ID WHERE USER_ID = P_OLD_USER_ID 
+            AND VISIT_ID NOT IN ( SELECT VISIT_ID FROM VISITS_HAS_USERS WHERE USER_ID = P_NEW_USER_ID );
+          DELETE FROM VISITS_HAS_USERS WHERE USER_ID = P_OLD_USER_ID;
+
+          UPDATE FAP_SECRETARIES SET USER_ID = P_NEW_USER_ID WHERE USER_ID = P_OLD_USER_ID 
+            AND FAP_ID NOT IN ( SELECT FAP_ID FROM FAP_SECRETARIES WHERE USER_ID = P_NEW_USER_ID ) ;
+          DELETE FROM FAP_SECRETARIES WHERE USER_ID = P_OLD_USER_ID;
+
+          UPDATE FAP_ASSIGNMENTS SET FAP_MEMBER_USER_ID = P_NEW_USER_ID WHERE FAP_MEMBER_USER_ID = P_OLD_USER_ID AND 
+         (CAST(PROPOSAL_PK AS TEXT) || CAST(FAP_ID AS TEXT)) NOT IN ( SELECT CAST(PROPOSAL_PK AS TEXT) || CAST(FAP_ID AS TEXT) as FA_KEY 
+          FROM FAP_ASSIGNMENTS WHERE fap_member_user_id = P_NEW_USER_ID );
+          DELETE FROM FAP_ASSIGNMENTS WHERE FAP_MEMBER_USER_ID = P_OLD_USER_ID AND 
+          (CAST(PROPOSAL_PK AS TEXT) || CAST(FAP_ID AS TEXT)) IN ( SELECT CAST(PROPOSAL_PK AS TEXT) || CAST(FAP_ID AS TEXT) as FA_KEY 
+          FROM FAP_ASSIGNMENTS WHERE fap_member_user_id = P_NEW_USER_ID );
+
+          DELETE FROM USERS WHERE USER_ID = P_OLD_USER_ID;
+
+          RETURN; 
+          COMMIT;
+          EXCEPTION WHEN OTHERS THEN
+              RAISE EXCEPTION 'ROLLBACK UPDATE_USER_ID FUNC WITH OLD USER ID: % AND NEW USER ID: %', P_OLD_USER_ID, P_NEW_USER_ID;
+              ROLLBACK;
+      END;
+      $$;
+    END;
+	END IF;
```

Comment:
Looks like there's tabs instead of spaces that're causing extra indents on the github comparison for the IF/END IF lines